v0/util/index.js

```diff
@@ -0,0 +1,351 @@
+    type: "REST",
+    method: "POST",
+    endpoint: "",
+    headers: {},
+    params: {},
+    body: {
+      JSON: {},
+      XML: {},
+      FORM: {}
+    },
+    files: {}
+  };
+};
+
+// ========================================================================
+// TRANSFORMER UTILITIES
+// ========================================================================
+const MESSAGE_MAPPING = JSON.parse(
+  fs.readFileSync(path.resolve(__dirname, `./data/GenericFieldMapping.json`))
+);
+
+// Get the IP address from the message.
+// NEED TO DEPRECATE
+const getParsedIP = message => {
+  if (message.context && message.context.ip) {
+    return message.context.ip;
+  }
+  return message.request_ip;
+};
+
+// Important !@!
+// create the mappingConfig from data JSONs for destinations
+const getMappingConfig = (config, dir) => {
+  const mappingConfig = {};
+  const categoryKeys = Object.keys(config);
+  categoryKeys.forEach(categoryKey => {
+    const category = config[categoryKey];
+    mappingConfig[category.name] = JSON.parse(
+      fs.readFileSync(path.resolve(dir, `./data/${category.name}.json`))
+    );
+  });
+  return mappingConfig;
+};
+
+// NEED a better way to handle it. Only used in Autopilot and Intercom
+// NEED TO DEPRECATE
+const updatePayload = (currentKey, eventMappingArr, value, payload) => {
+  eventMappingArr.forEach(obj => {
+    if (obj.rudderKey === currentKey) {
+      set(payload, obj.expectedKey, value);
+    }
+  });
+  return payload;
+};
+
+// Important !@!
+// - get value from a list of sourceKeys in precedence order
+// - get value from a string key
+const getValueFromMessage = (message, sourceKey) => {
+  if (Array.isArray(sourceKey) || sourceKey.length === 0) {
+    // got the possible sourceKeys
+    for (let index = 0; index < sourceKey.length; index += 1) {
+      const val = get(message, sourceKey[index]);
+      if (val) {
+        // return only if the value is valid.
+        // else look for next possible source in precedence
+        return val;
+      }
+    }
+  } else if (typeof sourceKey === "string") {
+    // got a single key
+    // - we don't need to iterate over a loop for a single possible value
+    return get(message, sourceKey);
+  } else {
+    // wrong sourceKey type. abort
+    // DEVELOPER ERROR
+    throw new Error("Wrong sourceKey type or blank sourceKey array");
+  }
+  return null;
+};
+
+// format the value as per the metadata values
+// Expected metadata keys are: (according to precedence)
+// - - type, typeFormat: expected data type and its format
+// - - template : need to have a handlebar expression {{value}}
+const handleMetadataForValue = (value, metadata) => {
+  let formattedVal = value;
+  const { type, typeFormat, template } = metadata;
+
+  // handle type and format
+  // skipping check for typeFormat to support default for each type
+  if (type) {
+    switch (type) {
+      case "timestamp":
+        formattedVal = formatTimeStamp(formattedVal, typeFormat);
```

Comment:
as there could be multiple timestamp formats, we may either put different type for them in metadata or put a subtype as "UTC" and based on that call different methods or pass it in the same method as argument.